edi_purchase_edifact_oca/wizard/purchase_order_import.py

```diff
@@ -0,0 +1,681 @@
+                    "The quantity delivered for product "
+                    "<a href=# data-oe-model=product.product "
+                    f"data-oe-id={order_line.product_id.id}>"
+                    f"{order_line.product_id.name}</a> "
+                    "is less than or equal to the quantity received."
+                )
+            )
+        if total_qty_update > 0:
+            for move in moves:
+                if total_qty_update <= 0:
+                    break
+                available_qty = move.product_uom_qty - move.quantity
+
+                if available_qty > 0:
+                    if total_qty_update >= available_qty:
+                        move.quantity = move.product_uom_qty
+                        total_qty_update -= available_qty
+                    else:
+                        move.quantity += total_qty_update
+                        total_qty_update = 0
+
+                    if not picking_dict.get(move.picking_id, False):
+                        picking_dict[move.picking_id] = []
+                    picking_dict[move.picking_id].append(move.id)
+
+            self._update_qty_done_package(moves)
+
+            # Create new stock.move if still available
+            if total_qty_update > 0:
+                new_move = moves[-1].copy(
+                    {
+                        "product_uom_qty": total_qty_update,
+                        "quantity": total_qty_update,
+                        "state": "draft",
+                    }
+                )
+                new_move = self._update_qty_done_package(new_move._action_confirm())
+                new_move._action_assign()
+
+            # Return the pickings with updated moves to print the message
+            return picking_dict
+        return False
+
+    @api.model
+    def _prepare_update_order_vals(self, parsed_order):
+        bdio = self.env["business.document.import"]
+        partner = bdio._match_partner(
+            parsed_order["company"],
+            parsed_order["chatter_msg"],
+            partner_type="customer",
+        )
+        vals = {"partner_id": partner.id}
+        return vals
+
+    def update_purchase_order(self, order, parsed_order):
+        self.ensure_one()
+        bdio = self.env["business.document.import"]
+        currency = bdio._match_currency(
+            parsed_order.get("currency"), parsed_order["chatter_msg"]
+        )
+        if currency != order.currency_id:
+            raise UserError(
+                _(
+                    "The currency of the imported order {old} is different from "
+                    "the currency of the existing order {new}"
+                ).format(
+                    old=currency.name,
+                    new=order.currency_id.name,
+                )
+            )
+        vals = self._prepare_update_order_vals(parsed_order)
+        if vals:
+            order.write(vals)
+        number_line_updated = self.update_order_lines(parsed_order, order)
```

Comment:
In v18, by default the quantity of stock.picking equals `product_uom_qty`. Therefore, we will set it to `0` and update it only on the lines that are actually received.
```suggestion
        for move in order.order_line.mapped("move_ids"):
            if (
                move.state not in ("done", "cancel", "draft")
                and move.product_uom_qty == move.quantity
            ):
                move.quantity = 0
                
        number_line_updated = self.update_order_lines(parsed_order, order)
```